### ISSUE TITLE
Move app icons closer together

### ISSUE THREAD

**Issue by Edenhofer** [micheleg/dash-to-dock#341]
Currently the dock takes quite a chunk of space from the bar where it is positioned. However most of it is consumed by gaps between the icons and not by actual "clickable" content. An option to move the icons closer together (make the dock more dense) would be great.

**Comment by micheleg** [comment]
Hi, how is this different from the effect of activating the "shrink the dash" option in the Appearance settings? Are you suggesting an even more condensed layout?

**Comment by Edenhofer** [comment]
Yes exactly. Though shrinking the dock is noticably it does not provide a really contensed dock.

------------------------------

#### Here is the normal dash-to-dock bar

![normal](https://cloud.githubusercontent.com/assets/7541458/16501159/60d59690-3f08-11e6-8b2f-d9be41bf181a.png)

------------------------------

#### Here a downsized one

![downsize](https://cloud.githubusercontent.com/assets/7541458/16501224/adfc0b52-3f08-11e6-9430-d227b17a2ffb.png)

------------------------------

#### and here is e.g. docky

![docky](https://cloud.githubusercontent.com/assets/7541458/16501235/b7440638-3f08-11e6-94db-da080d6d9c11.png)

**Comment by micheleg** [comment]
It's not clear to me fro your screenshot if the shrinking is happening. With the default theme the shrink function save quite a bit of space. Can you check with the default theme? (I'm trying to see if there is a problem with 3rd parties themes) 

I agree that additional space can be saved by making the launcher even more condensed. For instance if you add something like this in `~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/stylesheet.css`  and restart the shell you should get a much more condensed dock.
```
#dashtodockContainer .overview-icon {
padding:0px;
}
```

I could provide an option like wide - normal- condensed  dock, but I'm concern I would not be able to make it easily compatible with all themes.

**Comment by Edenhofer** [comment]
I checked with the default theme and the "bug" persisted. The marginal difference is the same as the one which is depicted. Shrinking the dock mainly influences the height.

Your quick-fix however worked perfectly! A little too good to be quite honest. Now the icons are extremely condensted with no space in between, the dots (window indicator) beneath the icons are now covered by the icon. A slider or fine-grained drop-down menu would be a perfect solution in my eyes.

**Comment by Filbuntu** [comment]
This feature would really helpful! 
I could tweak it with the addition of the code you mentioned, but not everyone wants to or is able to do it, therefore an option in the settings would be great.
I put the padding to 3px and was happy with the outcome.
Thanks a lot for the work on this Dock, much appreciated!

**Comment by Filbuntu** [comment]
PS: Although the tweak works, when there are updates (not sure of which package), the manual addition was deleted:
```
#dashtodockContainer .overview-icon {
padding:0px;
}
```
I had to add it to the file `~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/stylesheet.css ` again.
A more permanent change option would be much appreciated, thanks!

**Comment by corebots** [comment]
Is there a way to increase the space between the running-indicators and the icons after reducing the overall padding?

**Comment by Filbuntu** [comment]
I would still be happy to get  a more permanent change option. Thanks!
